### PR TITLE
Added an initial GPS position.

### PR DESCRIPTION
**Pull request by imisaacwu**

### CODE REVIEW

File: Assets/Scripts/GpsSensor.cs [hunk 1]
```diff
@@ -51,21 +56,35 @@ private IEnumerator StreamPosition()
 
     private void ReportPosition()
     {
-        // Use double precision since geographic degrees are very large.
-        double latitude = CartesianToGeographic(transform.position.z + _noise * Utilities.GaussianRandom());
-        double longitude = CartesianToGeographic(transform.position.x + _noise * Utilities.GaussianRandom());
+        // Assuming rover initializes facing north, i.e. z+ (forward) is north & x+ (right) is east
+        double[] GPS = metersToGPS(new double[] {
+            transform.position.z + _noise * Utilities.GaussianRandom(),
+            transform.position.x + _noise * Utilities.GaussianRandom()});
```
Comment (abhaybd) [inline Assets/Scripts/GpsSensor.cs]: How does this assume the initial orientation of the rover? Seems like it's just using information about the unity coordinate system, and we've defined north to be +z. So I'd say just leave a comment saying that +z is north and +x is east.

File: Assets/Scripts/GpsSensor.cs [hunk 2]
```diff
@@ -51,21 +56,35 @@ private IEnumerator StreamPosition()
...
-        return Mathf.Rad2Deg * meters / EarthRadius;
+        // +x is +lat, +y is -lon
+        double semiMajorAxis = 6378137.0;
+        double semiMinorAxis = 6356752.314245;
+        double phi = Math.PI * initLat / 180.0;
+        // Square Eccentricity
+        double eSq = 1 - (Math.Pow(semiMinorAxis, 2)) / (Math.Pow(semiMajorAxis, 2));
+        double var = 1 - eSq * Math.Pow(Math.Sin(phi), 2);
+        double metersPerDegLon = (Math.PI * semiMajorAxis * phi) / (180.0 * Math.Sqrt(var));
+        double metersPerDegLat = (Math.PI * semiMajorAxis * (1 - eSq)) / (180.0 * Math.Pow(var, 1.5));
```
Comment (abhaybd) [inline Assets/Scripts/GpsSensor.cs]: We should leave documentation explaining these magic numbers and math. See how we did it [in the rover code](https://github.com/huskyroboticsteam/Resurgence/blob/master/src/gps/gps_util.cpp).

Also, note that we're use the WGS84 datum, which is the source of those minor/major axes magic numbers.

File: Assets/Scripts/GpsSensor.cs [hunk 3]
```diff
@@ -51,21 +56,35 @@ private IEnumerator StreamPosition()
...
+        double[] gps = new double[2];
```
Comment (abhaybd) [inline Assets/Scripts/GpsSensor.cs]: Are we doing anything with this array after declaring it? Looks like we should delete it.

File: Assets/Scripts/GpsSensor.cs [hunk 4]
```diff
@@ -51,21 +56,35 @@ private IEnumerator StreamPosition()
...
+        double degDiffLat = offset[0] / metersPerDegLat;
+        double degDiffLon = -offset[1] / metersPerDegLon;
```
Comment (abhaybd) [inline Assets/Scripts/GpsSensor.cs]: Why are we negating `degDiffLon`?

File: Assets/Scripts/GpsSensor.cs [hunk 5]
```diff
@@ -22,6 +23,10 @@ public class GpsSensor : MonoBehaviour
     private float _noise;
     [SerializeField]
     private float _reportPeriod;
+    [SerializeField]
+    private double initLat = 47.653749871465955;
+    [SerializeField]
+    private double initLong = -122.30429294489063;
```
Comment (abhaybd) [inline Assets/Scripts/GpsSensor.cs]: Should leave a comment somewhere saying what these coordinates point to.

**Review by abhaybd — CHANGES_REQUESTED** [review]
Mostly good, just some notes about documentation and some other stuff.

File: Assets/Scripts/GpsSensor.cs (hunk 4 above)
Comment (imisaacwu) [reply Assets/Scripts/GpsSensor.cs]: The original code negated it, is this just funky coordinate handling?
gps_util.cpp @ 62
```
gpscoords_t GPSToMetersConverter::metersToGPS(const point_t& pos) const {
	// +x is +lat, +y is -lon
	double degDiffLat = pos(0) / metersPerDegLat;
	double degDiffLon = -pos(1) / metersPerDegLon;
	return {origin.lat + degDiffLat, origin.lon + degDiffLon};
}
```

Comment (imisaacwu) [reply Assets/Scripts/GpsSensor.cs]: Never mind, the longitude shouldn't be negated based on our starting position.

**Review by abhaybd — APPROVED** [review]
Looks good!

**Pull request merged**